### PR TITLE
Added BinaryWriter Span-based APIs

### PR DESCRIPTION
**Pull request by jp2masa**
Added tests for BinaryWriter Span-based APIs and exposed them from System.IO contract.
Connected to https://github.com/dotnet/coreclr/pull/13378

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by jp2masa** [comment]
@dotnet-bot test this please

**Comment by jp2masa** [comment]
@dotnet-bot test this please

**Comment by jp2masa** [comment]
@dotnet-bot
test NETFX x86 Release Build please
test OSX x64 Debug Build please
test UWP CoreCLR x64 Debug Build please
test UWP NETNative x86 Release Build please

**Comment by jp2masa** [comment]
@dotnet-bot
test NETFX x86 Release Build please
test UWP CoreCLR x64 Debug Build please
test UWP NETNative x86 Release Build please

File: src/System.IO/tests/BinaryWriter/BinaryWriter.WriteByteCharTests.cs [hunk 1]
```diff
@@ -558,5 +558,40 @@ public void BinaryWriter_WriteCharArrayTest3()
             mstr.Dispose();
             dw2.Dispose();
         }
+
+        [Fact]
+        public void BinaryWriter_WriteSpan()
```
Comment (stephentoub) [inline src/System.IO/tests/BinaryWriter/BinaryWriter.WriteByteCharTests.cs]: You need to put the test in a file that's only included in the netcoreapp build, e.g. conditioned like this file:
https://github.com/jp2masa/corefx/blob/6546a430e9d05f52157c0a426fbb1c70d3662835/src/System.IO/tests/System.IO.Tests.csproj#L43

**Comment by jp2masa** [comment]
@dotnet-bot test Windows x64 Debug Build please

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**